VirtualAssistantBusinessLogic/VirtualAssistantBusinessLogicTests/SparQL/SparQLWhereTests.cs

```diff
@@ -0,0 +1,114 @@
+﻿using NSubstitute;
+using NUnit.Framework;
+using System;
+using System.Collections.Generic;
+using VirtualAssistantBusinessLogic.KnowledgeGraph;
+using VirtualAssistantBusinessLogic.SparQL;
+
+namespace VirtualAssistantBusinessLogicTests.SparQL
+{
+    [TestFixture]
+    public class SparQLWhereTests
+    {
+        private SparQLSelect subSparQLSelect;
+        private ISPOEncoder subSPOEncoder;
+
+        [SetUp]
+        public void SetUp()
+        {
+            EncodedSPO output = new($"?s1 ?p \"SUBJECT\"@en . ", $"?s1");
+            subSPOEncoder = Substitute.For<ISPOEncoder>();
+            subSPOEncoder
+                .EncodeSubject(Arg.Any<string>())
+                .Returns(output);
+
+            subSparQLSelect = Substitute.For<SparQLSelect>(subSPOEncoder);
+        }
+
+        private SparQLWhere CreateSparQLWhere()
+        {
+            return new SparQLWhere(
+                this.subSparQLSelect);
+        }
+
+        [Test]
+        public void SubjectIs_HasEncodedSPO_SubjectStringCorrect()
+        {
+            // Arrange
+            SparQLWhere sparQLWhere = this.CreateSparQLWhere();
+            sparQLWhere.EncodedSPOs.Add("TEST", new EncodedSPO("TRIPLET", "NAME"));
+            string subject = "TEST";
+
+
+            // Act
+            _ = sparQLWhere.SubjectIs(
+                subject);
+
+            // Assert
+            Assert.AreEqual("NAME", sparQLWhere.SubjectString);
+        }
+
+        [Test]
+        public void SubjectIs_NoEncodedSPO_SubjectStringCorrect()
+        {
+            // Arrange
+            SparQLWhere sparQLWhere = this.CreateSparQLWhere();
+            string subject = "TEST";
+
+
+            // Act // Assert
+            Assert.Throws<KeyNotFoundException>(
+                () => sparQLWhere.SubjectIs(subject)
+                );
+        }
```

Comment:
Den her test skal nok slettes, for det den tester er ikke funktionalitet vi skal have (subject har ikke behøves at være encoded)